app/Http/Controllers/LogController.php

```
@@ -273,6 +307,61 @@ public function getLogCreateByUser(Request $request)
+        } catch (TokenExpiredException $e) {
+            return response()->json(['status'=> 'error', 'message' => 'Token has expired.'], 401);
+        } catch (TokenInvalidException $e) {
+            return response()->json(['status'=> 'error', 'message' => 'Token is invalid.'], 401);
+        } catch (JWTException $e) {
+            return response()->json(['status'=> 'error', 'message' => 'Token is absent or could not be parsed.'], 401);
+        } catch (\Exception $e) {
+            return response()->json(['status' => 'error', 'message' => 'Could not retrieve logs. ' . $e->getMessage()], 500);
+        }
+    }
+
+    public function getLogBySoftware(Request $request, $softwareId)
```

PHPMD (2.15.0)

Avoid unused parameters such as '$request'. (Unused Code Rules)

(UnusedFormalParameter)